net/http/src/THttpLongPollEngine.cxx

```diff
@@ -104,8 +177,17 @@ Bool_t THttpLongPollEngine::PreviewData(THttpCallArg &arg)
    }
 
    if (fQueue.size() > 0) {
-      arg.SetContentType("text/plain");
-      arg.SetContent(fQueue.front().c_str());
+      QueueItem &item = fQueue.front();
+      if (item.buf) {
+         arg.SetContentType("application/x-binary");
+         arg.SetBinData((void *)item.buf, item.len);
+         item.reset_buf(); // forget memory
+         if (!fRaw && !item.msg.empty())
```

Comment:
Why is the condition here different from that at line 118? In general I would hope that this repeated logic (queue? If not, raw?) can becentralized, and invoked from `SendCharStar()` and friends.

Reply:
It is not exactly the same situation. At line 118 I send binary plus text header.
Here I can have two variants - binary buffer with or without header. And if I use normal (not raw) mode, I can provide extra text message as custom http header field.

Raw mode means, that all kind of information should be packed into reply body, all kind of http headers are ignored (a case in QWebEngine)